cmd/libs/go2idl/make-client/generators/client-generator.go

```diff
@@ -0,0 +1,283 @@
+/* Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package generators has the generators for the set-gen utility.
+package generators
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/args"
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+
+	"github.com/golang/glog"
+)
+
+// NameSystems returns the name system used by the generators in this package.
+func NameSystems() namer.NameSystems {
+	return namer.NameSystems{
+		"public":  namer.NewPublicNamer(0),
+		"private": namer.NewPrivateNamer(0),
+		"raw":     namer.NewRawNamer("", nil),
+	}
+}
+
+// DefaultNameSystem returns the default name system for ordering the types to be
+// processed by the generators in this package.
+func DefaultNameSystem() string {
+	return "public"
+}
+
+// Packages makes the sets package definition.
+func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
+	boilerplate, err := arguments.LoadGoBoilerplate()
+	if err != nil {
+		glog.Fatalf("Failed loading boilerplate: %v", err)
+	}
+
+	// @lavalamp: This is quite hacky. Any suggestions?
```

Comment:
Just to ensure that we are on the same page. You would like to generate clients for all "internal" types (i.e. api.Pod, api.ReplicationController, ..., extensions.Deployment, extensions.Job, ...).

So what I would do is:
- in the main binary overwrite the InputDirs:
https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/set-gen/main.go#L39
- it should container list of all packages, where your types are defined, i.e.:
k8s.io/kubernetes/pkg/api
k8s.io/kubernetes/pkg/apis/extensions
(possibly some others)
- for every type you would like to generate the types, add the following comment line next to it:
// +genclient=true
- then here, iterate over all types in the context and filter only those that have the genclient "bit" set

This exact pattern is used in set-gen example, in particular the filtering is done here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/set-gen/generators/sets.go#L115